gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -291,41 +403,30 @@ impl ILnRpcClient for GatewayLdkClient {
                 max_channel_saturation_power_of_half: None,
             }),
         ) {
-            Ok(payment_id) => payment_id,
-            Err(e) => {
-                return Err(LightningRpcError::FailedPayment {
-                    failure_reason: format!("LDK payment failed to initialize: {e:?}"),
-                });
-            }
+            Some(Err(LightningRpcError::FailedPayment {
+                failure_reason: format!("LDK payment failed to initialize: {err:?}"),
+            }))
+        } else {
+            None
         };
 
-        // TODO: Find a way to avoid looping/polling to know when a payment is
-        // completed. `ldk-node` provides `PaymentSuccessful` and `PaymentFailed`
-        // events, but interacting with the node event queue here isn't
-        // straightforward.
-        loop {
-            if let Some(payment_details) = self.node.payment(&payment_id) {
-                match payment_details.status {
-                    PaymentStatus::Pending => {}
-                    PaymentStatus::Succeeded => {
-                        if let PaymentKind::Bolt11 {
-                            preimage: Some(preimage),
-                            ..
-                        } = payment_details.kind
-                        {
-                            return Ok(PayInvoiceResponse {
-                                preimage: Preimage(preimage.0),
-                            });
-                        }
-                    }
-                    PaymentStatus::Failed => {
-                        return Err(LightningRpcError::FailedPayment {
-                            failure_reason: "LDK payment failed".to_string(),
-                        });
-                    }
-                }
-            }
-            fedimint_core::runtime::sleep(Duration::from_millis(100)).await;
+        if let Some(response) = response_or {
```

Comment:
Add comment here that this semantically means the payment has already been completed and we can just return the retrieved `PayInvoiceResponse`

Reply:
Done